src/Snowflake/Helper.php

```diff
@@ -94,30 +89,38 @@ public static function generateRandomString(int $length = 10): string
         return $randomString;
     }
 
+    /**
+     * @param GrantToRole[]|FutureGrantToRole[] $schemasGrants
+     * @return GrantToRole[]|FutureGrantToRole[]
+     */
     public static function filterSchemaGrants(string $database, string $schemaName, array $schemasGrants): array
     {
         return array_filter(
             $schemasGrants,
-            function (array $v) use ($database, $schemaName) {
-                $validSchema = [
-                    sprintf('%s.%s', $database, $schemaName),
-                    sprintf('%s.%s', $database, QueryBuilder::quoteIdentifier($schemaName)),
-                    sprintf('%s.%s', QueryBuilder::quoteIdentifier($database), $schemaName),
-                    sprintf(
-                        '%s.%s',
-                        QueryBuilder::quoteIdentifier($database),
-                        QueryBuilder::quoteIdentifier($schemaName)
-                    ),
-                ];
-                return in_array($v['name'], $validSchema);
+            function (GrantToRole|FutureGrantToRole $v) use ($database, $schemaName) {
+                $validSchema = Helper::generateFormattedQuoteCombinations($database, $schemaName);
+                return in_array($v->getName(), $validSchema);
             }
         );
     }
 
-    public static function removeStringFromEnd(string $haystack, string $needle): string
+    public static function generateFormattedQuoteCombinations(string ...$inputArray): array
     {
-        $needle = preg_quote($needle, '/');
-        return (string) preg_replace("/$needle$/", '', $haystack);
+        $ret = [];
+        for ($b = 0; $b < pow(count($inputArray), 2); $b++) {
+            $binNum = str_pad(decbin($b), count($inputArray), '0', STR_PAD_LEFT);
+            if (strlen($binNum) === count($inputArray)) {
+                $variables = [];
+                foreach ($inputArray as $k => $item) {
+                    $variables[] = $binNum[$k] === '1' ? Helper::quoteIdentifier($item) : $item;
+                }
+                $ret[] = sprintf(
+                    implode('.', array_pad([], count($inputArray), '%s')),
+                    ...$variables
+                );
+            }
+        }
+        return $ret;
```

Comment:
🤯 
tohle nejdela 
```
$needle = preg_quote($needle, '/');
return (string) preg_replace("/$needle$/", '', $haystack);
```
vit?  tak proc to zmenime?

Reply:
vyřešeno na cellu